lib/kennel/config.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+# The options which affect the running of Kennel.
+#
+# The options are:
+#
+# * `out` - where kennel prints output (an `IO`; default: `$stdout`)
+# * `err` - where kennel prints errors (an `IO`; default: `$stderr`)
+# * `strict_imports` - whether kennel should abort if asked to create a part, and the part has an `id:`,
+#   but there is no such object in Datadog (boolean; default: true)
+#
+# To build the default config:
+#
+#   config = Kennel::Config.new
+#
+# To build a config which reads its settings from various ENV variables:
+#
+#   config = Kennel::Config.from_env
+#
+# To build a config from a hash, or from another config:
+#
+#   config = Kennel::Config.new(from)
+#
+# A block can be passed to `.new` or `.from_env`. If a block is given,
+# then after any settings have been copied from `from` (if given),
+# a mutable config object will be yielded to the block, so that
+# the config can be set up:
+#
+#   config = Kennel::Config.new do |c|
+#     c.strict_imports = false
+#   end
+#
+# Once built, a config object is immutable.
```

Comment:
not a fan of that, makes testing/experimenting harder 🤷

Reply:
I have the exact opposite viewpoint. Mutability makes testing harder. Don't like the config? Make a new one.

Reply:
if it's a global object then yeah, if it's per-instance then I don't really see the benefit since there would be a new one per test anyways
(no big deal either way)